### PR TITLE
[fix](iceberg) update iceberg docs and add credential properties

### PR DESCRIPTION
**Pull request by wsjz**
# Proposed changes

Update iceberg docs 
Add new s3 credential and properties

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 34.62 seconds
 load time: 470 seconds
 storage size: 17170763067 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230206043021_clickbench_pr_90638.html

File: docs/en/docs/lakehouse/multi-catalog/iceberg.md [hunk 1]
```diff
@@ -52,6 +54,50 @@ CREATE CATALOG iceberg PROPERTIES (
 );
 ```
 
+### Iceberg Native Catalog
+
+<version since="dev">
+
+Access metadata with the iceberg API. The Hive, REST, Glue and other services can serve as the iceberg catalog.
+
```
Comment (morningman) [inline docs/en/docs/lakehouse/multi-catalog/iceberg.md:62]: missing the end tag of `<version>`
Add here: `</version>`

File: docs/en/docs/lakehouse/multi-catalog/iceberg.md [hunk 2]
```diff
@@ -52,6 +54,50 @@ CREATE CATALOG iceberg PROPERTIES (
...
+    'type'='iceberg',
+    'iceberg.catalog.type'='rest',
+    'uri' = 'http://172.21.0.1:8181',
+);
+```
+
+If you want to use S3 storage, the following properties need to be set.
+
+```
+"AWS_ACCESS_KEY" = "username"
```
Comment (morningman) [inline docs/en/docs/lakehouse/multi-catalog/iceberg.md:94]:
```suggestion
"AWS_ACCESS_KEY" = "ak"
```

File: docs/en/docs/lakehouse/multi-catalog/iceberg.md [hunk 3]
```diff
@@ -52,6 +54,50 @@ CREATE CATALOG iceberg PROPERTIES (
...
+"AWS_SECRET_KEY" = "password"
```
Comment (morningman) [inline docs/en/docs/lakehouse/multi-catalog/iceberg.md:95]:
```suggestion
"AWS_SECRET_KEY" = "sk"
```

File: docs/zh-CN/docs/lakehouse/multi-catalog/iceberg.md [hunk 4]
```diff
@@ -54,6 +54,8 @@ CREATE CATALOG iceberg PROPERTIES (
 
 ### 基于Iceberg API创建Catalog
 
+<version since="dev">
```
Comment (morningman) [inline docs/zh-CN/docs/lakehouse/multi-catalog/iceberg.md:57]: ditto

File: docs/en/docs/lakehouse/multi-catalog/iceberg.md [hunk 5]
```diff
@@ -52,6 +54,50 @@ CREATE CATALOG iceberg PROPERTIES (
...
+"AWS_REGION" = "region-name"
+"AWS_ENDPOINT" = "http://endpoint-uri"
+"AWS_CREDENTIALS_PROVIDER" = "provider-class-name" // Optional. The default credentials class is based on BasicAWSCredentials.
```
Comment (morningman) [inline docs/en/docs/lakehouse/multi-catalog/iceberg.md:98]: The default should be `com.amazonaws.glue.catalog.credentials.ConfigurationAWSCredentialsProviderFactory`

File: fe/fe-core/src/main/java/org/apache/doris/catalog/S3Resource.java [hunk 6]
```diff
@@ -222,41 +226,36 @@ protected void getProcNodeData(BaseProcResult result) {
...
+        s3Properties.put("fs.s3.impl", S3AFileSystem.class.getName());
 
-        if (Boolean.valueOf(properties.getOrDefault(S3Resource.USE_PATH_STYLE, "false")).booleanValue()) {
-            s3Properties.put("fs.s3a.path.style.access", "true");
-        } else {
-            s3Properties.put("fs.s3a.path.style.access", "false");
-        }
+        s3Properties.put(Constants.PATH_STYLE_ACCESS, properties.getOrDefault(S3Resource.USE_PATH_STYLE, "false"));
+        s3Properties.put(Constants.AWS_CREDENTIALS_PROVIDER, properties.getOrDefault(S3Resource.S3_CREDENTIALS_PROVIDER,
+                TemporaryAWSCredentialsProvider.class.getName()));
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/S3Resource.java:255]: default should be `com.amazonaws.glue.catalog.credentials.ConfigurationAWSCredentialsProviderFactory`

File: fe/fe-core/src/main/java/org/apache/doris/catalog/S3Resource.java [hunk 7]
```diff
@@ -222,41 +226,36 @@ protected void getProcNodeData(BaseProcResult result) {
...
         if (properties.containsKey(S3Resource.S3_TOKEN)) {
-            s3Properties.put("fs.s3a.session.token", properties.get(S3_TOKEN));
-            s3Properties.put("fs.s3a.aws.credentials.provider",
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/S3Resource.java:256]: Add this back

File: docs/en/docs/lakehouse/multi-catalog/iceberg.md (hunk 5 above)
Comment (wsjz) [reply docs/en/docs/lakehouse/multi-catalog/iceberg.md:98]: use ConfigurationAWSCredentialsProviderFactory for glue

File: fe/fe-core/src/main/java/org/apache/doris/catalog/S3Resource.java (hunk 6 above)
Comment (wsjz) [reply fe/fe-core/src/main/java/org/apache/doris/catalog/S3Resource.java:255]: current use ConfigurationAWSCredentialsProviderFactory for glue

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**